relay/cache/fifo-cache.go

```diff
@@ -0,0 +1,84 @@
+package cache
+
+import (
+	"errors"
+	"github.com/Layr-Labs/eigenda/common/queue"
+)
+
+var _ Cache[string, string] = &FIFOCache[string, string]{}
+
+// FIFOCache is a cache that evicts the least recently added item when the cache is full. Useful for situations
+// where time of addition is a better predictor of future access than time of most recent access.
+type FIFOCache[K comparable, V any] struct {
```

Comment:
No, it's just that FIFO is a lot easier to implement. I think it's premature to declare either to be faster at the moment, since we don't know for certain what the access pattern is going to be.